src/zcl_abapgit_repo_online.clas.abap

```diff
@@ -321,41 +311,28 @@ CLASS zcl_abapgit_repo_online IMPLEMENTATION.
                                      IMPORTING ev_branch        = lv_branch
                                                et_updated_files = lt_updated_files ).
 
-    IF io_stage->get_branch_sha1( ) = get_sha1_local( ).
-* pushing to the branch currently represented by this repository object
-      mv_branch = lv_branch.
-      set( iv_sha1 = lv_branch ).
-    ELSE.
-      refresh( ).
-    ENDIF.
-
+    refresh( ).
```

Comment:
The code deleted above is a performance optimization to avoid calling `refresh()` after each push.
Suggest just deleting line 327

Reply:
Hm, ok, but how should the condition in 324 look like? Method get_sha1_local( ) doesn't exist anymore. Do we need the refresh( ) at all?
https://github.com/larshp/abapGit/blob/10a14dd3879e13375ee3ca6e35cf32a067be0674/src/zcl_abapgit_repo_online.clas.abap#L324

Reply:
try get_sha1_remote() instead of get_sha1_local()

I dont think refresh() is needed, but lets move that to a different issue/PR

Reply:
done